gunrock/cuda/launch_box.hxx

```diff
@@ -12,44 +12,148 @@
 
 #include <gunrock/cuda/device_properties.hxx>
 
+// Don't need SM_TAG right now, but will in the future
+#ifdef __CUDA_ARCH__
+#if   __CUDA_ARCH__ >= 860
+  #define SM_TAG   sm_86
+  #define ARCH_TAG ampere
+#elif __CUDA_ARCH__ >= 800
+  #define SM_TAG   sm_80
+  #define ARCH_TAG ampere
+#elif __CUDA_ARCH__ == 750
+  #define SM_TAG   sm_75
+  #define ARCH_TAG turing
+#elif __CUDA_ARCH__ >= 700
+  #define SM_TAG   sm_70
+  #define ARCH_TAG volta
+#elif __CUDA_ARCH__ == 620
+  #define SM_TAG   sm_62
+  #define ARCH_TAG pascal
+#elif __CUDA_ARCH__ >= 610
+  #define SM_TAG   sm_61
+  #define ARCH_TAG pascal
+#elif __CUDA_ARCH__ >= 600
+  #define SM_TAG   sm_60
+  #define ARCH_TAG pascal
+#elif __CUDA_ARCH__ == 530
+  #define SM_TAG   sm_53
+  #define ARCH_TAG maxwell
+#elif __CUDA_ARCH__ >= 520
+  #define SM_TAG   sm_52
+  #define ARCH_TAG maxwell
+#elif __CUDA_ARCH__ >= 500
+  #define SM_TAG   sm_50
+  #define ARCH_TAG maxwell
+#elif __CUDA_ARCH__ == 370
+  #define SM_TAG   sm_37
+  #define ARCH_TAG kepler
+#elif __CUDA_ARCH__ >= 350
+  #define SM_TAG   sm_35
+  #define ARCH_TAG kepler
+#elif __CUDA_ARCH__ == 320
+  #define SM_TAG   sm_32
+  #define ARCH_TAG kepler
+#elif __CUDA_ARCH__ >= 300
+  #define SM_TAG   sm_30
+  #define ARCH_TAG kepler
+#else
+  #error "Gunrock only supports sm_30 and above"
+#endif  // __CUDA_ARCH__
+#else
+  // What should these macros be on the host side?
+  #define SM_TAG   sm_00
+  #define ARCH_TAG ampere
+#endif
```

Comment:
Why do we need these constants when we have `const` functions in device properties that you recently added? Can these not be determined using the names and arch major/minor?

Reply:
Each of the architectures need to be a struct template name in order for us to define them the way we want (`ampere<16, 64, 2>`). To do that we have to have macros for the architecture names.

I don't think there is a way to determine them with the device properties functions since all I could do is get the architecture name as a string.